src/Symfony/Bundle/SecurityBundle/Security/FirewallContext.php

```diff
@@ -25,7 +25,7 @@ class FirewallContext
     private $exceptionListener;
     private $config;
 
-    public function __construct(array $listeners, ExceptionListener $exceptionListener = null, FirewallConfig $config = null)
+    public function __construct(/* iterable */ $listeners, ExceptionListener $exceptionListener = null, FirewallConfig $config = null)
```

Comment:
that would break laziness as the firewall uses `getListeners()` for iterating over listeners. Not sure what should be the condition for transforming to array, we did not in https://github.com/symfony/symfony/pull/21516 nor https://github.com/symfony/symfony/pull/21450